### DISCUSSION TITLE
Page API Routes - Cookies not Being Read

### DISCUSSION THREAD

**Discussion by tzarger** [vercel/next.js#34869] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.14.0
      npm: 8.3.1
      Yarn: 1.22.17
      pnpm: N/A
    Relevant packages:
      next: 12.1.1-canary.4
      react: 17.0.2
      react-dom: 17.0.2

### What browser are you using? (if relevant)

Chrome : Version 98.0.4758.109 (Official Build) (arm64)

### How are you deploying your application? (if relevant)

Local development and test production build locally

### Describe the Bug

Cookies in pages/api not working on getting.

pages/_middleware is working.

Running `apollo-server-micro` and a custom context and req, req can load cookies.  both req.cookies and req.headers?.cookie are empty {}

### Expected Behavior

Expect cookies to set and read.
```js
 const options: CookieSerializeOptions = {
    maxAge: MAX_AGE,
    path: "/",
    sameSite: "strict",
    httpOnly: true,
    secure: false,
    domain: "localhost",
  };
  res.setHeader("Set-Cookie", serialize('token', data, options));
```

For Getting Token
```js
export function parseCookies(req: NextApiRequest) {
  // For API Routes we don't need to parse the cookies.
  if (req.cookies) return req.cookies;

  // For pages we do need to parse the cookies.
  const cookie = req.headers?.cookie;
  return parse(cookie || "");
}

export const getCookie = (cookieName: string, req: NextApiRequest): string | undefined => {
  const cookies = parseCookies(req);
  return cookies[cookieName];
};
```
getCookie return nothing as the parseCookie returns undefined.

### To Reproduce

I started with the `api-routes-apollo-server` example. If you upgrade apollo-server-micro to 3.0 or newer .. nether set or get cookies works. With < v3 of apollo-server-micro I can seem to at least set the cookie, but not read the cookie.

### Additional Note
In the Discord channel, there is another person who is not using GraphQL who also is not seeing cookies to be read in pages/api files. This is not directly specific to `apollo-server-micro`.

**Comment by tessamerrill** [comment]
Having a similar issue, but a bit backwards; `pages/_middleware.ts` can't see the cookies I'm setting on an API with the `Set-Cookie` header, however if I use a client-side cookie setting method, i.e. `js-cookies`, and set the cookie value to what the `Set-Cookie` value _would_ be setting the value to, then it basically works. However, this is a workaround I'd like to avoid.

**Comment by tzarger** [comment]
This seems to be an issue with getServerSide Props more specifically for me.
